backend/routes/portfolios.js

```diff
@@ -342,12 +341,196 @@ router.get("/portfolio/prediction/:id", async (req, res) => {
   const end = formatDate(nextDate);
   finnhubClient.earningsCalendar(
     { from: begin, to: end },
-    (error, data, response) => {
+    async (error, data, response) => {
       res.json(data);
+      for (let nextEarnings of data.earningsCalendar) {
+        // check that symbol exists in db:
+        const exists = await prisma.company.findUnique({
+          where: {
+            ticker: nextEarnings.symbol,
+          },
+        });
+        if (exists == null) {
+          continue;
+        }
+        await prisma.company.update({
+          where: {
+            ticker: nextEarnings.symbol,
+          },
+          data: {
+            UpcomingEarnings: [
+              nextEarnings.date,
+              String(nextEarnings.epsActual),
+              String(nextEarnings.epsEstimate),
+              String(nextEarnings.revenueActual),
+              String(nextEarnings.revenueEstimate),
+              nextEarnings.symbol,
```

Comment:
could this be a json/object instead of array? it is hard to keep track which element represents what.